Source/DataSources/KmlDataSource.js

```diff
@@ -1474,27 +1635,298 @@ define([
         console.log('KML - Unsupported feature: ' + node.localName);
     }
 
+    var RefreshMode = {
+        INTERVAL : 0,
+        EXPIRE : 1,
+        STOP : 2
+    };
+
+    var scratchCartographic = new Cartographic();
+    function computeViewRectangle(camera) {
+        if (!defined(camera)) {
+            return Rectangle.MAX_VALUE;
+        }
+
+        var region = camera.computeViewRegion();
+        if (!defined(region)) {
+            return Rectangle.MAX_VALUE;
+        }
+
+        var result;
+        var ellipsoid = Ellipsoid.WGS84;
+        var count = region.length;
+        for(var i=0;i<count;++i) {
+            ellipsoid.cartesianToCartographic(region[i], scratchCartographic);
+            if (!defined(result)) {
+                result = new Rectangle(scratchCartographic.longitude, scratchCartographic.latitude,
+                    scratchCartographic.longitude, scratchCartographic.latitude);
+            } else {
+                Rectangle.expand(result, scratchCartographic, result);
+            }
+        }
+
+        return result;
+    }
+
+    function makeQueryString(string1, string2, addQuestionMark) {
+        function cleanupString(s) {
```

Comment:
Move this function to be outside of `makeQueryString` otherwise you end up making a brand new function everytime this is called, which is bad practice.